api/src/main/java/org/openmrs/validator/DiagnosisValidator.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.openmrs.Diagnosis;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.APIException;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+/**
+ * Validates {@link Validator} objects
+ * This class ensures that the condition object is valid and properly structured
+ * 
+ * @since 2.2
+ */
+@Handler(supports = {Validator.class}, order = 50)
```

Comment:
Are you sure this is the class that is being validated?

Reply:
Changed this to the correct class